pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -675,21 +787,24 @@ func (nc *Controller) doEvictionPass() {
 				klog.Warningf("Failed to get Node %v from the nodeLister: %v", value.Value, err)
 			}
 			nodeUID, _ := value.UID.(string)
-			pods, err := listPodsFromNode(nc.kubeClient, value.Value)
-			if err != nil {
-				utilruntime.HandleError(fmt.Errorf("unable to list pods from node %q: %v", value.Value, err))
+			pods, ok := nc.nodeEvictionMap.deletePods(value.Value)
+			if !ok {
+				utilruntime.HandleError(fmt.Errorf("node %q eviction data doesn't exist", value.Value))
 				return false, 0
 			}
 			remaining, err := nodeutil.DeletePods(nc.kubeClient, pods, nc.recorder, value.Value, nodeUID, nc.daemonSetStore)
 			if err != nil {
+				// In case of error, we re-add pods back to nodeEvictionMap and rely on next iteration
+				// of doEvictionPass to retry the deletion if still needed.
+				_ = nc.nodeEvictionMap.addPods(value.Value, pods)
```

Comment:
Don't ignore what function returned silently.

Reply:
In case "false" is returned, we should probably log something like: "node X was unregistered in the meantime - skipping readding pods" or sth along those lines

Reply:
Done.